entree/backends/xilinxhls/writer.py

```diff
@@ -96,410 +100,119 @@ def write(ensemble_dict, cfg):
                     '{}/{}_reconfigurable_system/scripts/tcl/{}'.format(cfg['OutputDir'], cfg['ProjectName'], entry.name)
                 )
 
+    # Default clock:
+    if cfg.get('ClockPeriod', False) == False:
+        cfg['ClockPeriod'] = "5"
 
+    # Templates directory:
+    env = Environment(loader=FileSystemLoader(filedir))
```

Comment:
The environment should be initialized only once at package loading. Move as singleton to the `__init__` file